### PR TITLE
 Add more license files

### PR DESCRIPTION
**Pull request by lxn2**
Added more extension handlers to license_header.py. Made it verbose so we keep track of what we're whitelisting/skipping.

Added license headers to files with newly added extension headers.

### CODE REVIEW

**Comment by piiswrong** [comment]
scala test fails

**Comment by mli** [comment]
LGTM, thanks

File: cpp-package/example/lenet.cpp [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
 /*!
```
Comment (nswamy) [inline cpp-package/example/lenet.cpp]: remove empty comment lines line 20, 21

File: cpp-package/example/lenet_with_mxdataiter.cpp [hunk 2]
```diff
@@ -1,5 +1,23 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
 /*!
```
Comment (nswamy) [inline cpp-package/example/lenet_with_mxdataiter.cpp]: same as above

File: cpp-package/example/mlp.cpp [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
 /*!
- * Copyright (c) 2015 by Contributors
```
Comment (nswamy) [inline cpp-package/example/mlp.cpp]: same

**Comment by lxn2** [comment]
PR builds passed, got approval, will merge now.

**Pull request merged**